presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/QueryBuilder.java

```diff
@@ -102,7 +102,7 @@ public PreparedStatement buildSql(
 
         List<TypeAndValue> accumulator = new ArrayList<>();
 
-        List<String> clauses = toConjuncts(client, session, connection, tupleDomain, accumulator);
+        List<String> clauses = toConjuncts(session, connection, tupleDomain, accumulator);
         if (additionalPredicate.isPresent()) {
```

Comment:
What is `additionalPredicate`? Is it somehow connected with limit? If not, why don't you need it in delete?

Reply:
It is not. Connector when creates a splits then it can use additionalPredicate so it can run JDBC queries in parallel. One split where `key_column % 2 = 0` and other when `key_column % 2 = 1` where you have 2 splits.

Reply:
In this case you cannot delete something which does not exist (or can because it could somehow exist?), and it safe not to provide `additionalPredicate` for delete. However consider situation if you can not match indexes in underlying systems which had been created with such `additionalPredicate` conditions.

Reply:
This code is not executed for delete. `buildSql` is called only for `SELECT`. `DELETE` is using `buildDeleteSql` there is no `additionalPredicate`

Reply:
Yeah, I meant exactly that place (I started this discussion here because there is no `additionalPredicate` in code you added for `DELETE`).

Let me please clarify my comment.
Imagine situation when you have some data (key_column, year, name) in table `A`. 
And there are 2 splits: key_column % 2 = 0 and other when key_column % 2 = 1.
Usually user executes selects on A: `select name from A where year = <year>` which translates to two with splits:
```
select name from A where key_column % 2 = 0 and year = <year>
select name from A where key_column % 2 = 1 and year = <year>
``` 
To optimize these queries downstream system has `CREATE INDEX a_idx ON a (key_column, year)`
If you don't propagate `additionalPredicate` to `DELETE` this could end up with `a_idx` is not used for `DELETE`.
Could you please check this situation?

Reply:
> If you don't propagate additionalPredicate to DELETE this could end up with a_idx is not used for DELETE.

I don't care about it. It does not have to be well performant.

Notice that for this metadata delete there are no splits generated, hence there is no additional predicate nor even actual query execution in Presto. It all happens during the "planning".

Reply:
`additionalPredicate` is not about performance, but about correctness.
anyway, this is split's attribute, and no splits are involved at this point yet, afaict